src/me/ryanhamshire/PopulationDensity/DataStore.java

```diff
@@ -71,15 +60,15 @@
     //in-memory cache for messages
     private String[] messages;
 
-    //currently open region
-    private RegionCoordinates openRegionCoordinates;
+    //currently open regions
+    private ConcurrentHashMap<World, RegionCoordinates> openRegionCoordinates = new ConcurrentHashMap<>();
 
     //coordinates of the next region which will be opened, if one needs to be opened
-    private RegionCoordinates nextRegionCoordinates;
+    private ConcurrentHashMap<World, RegionCoordinates> nextRegionCoordinates = new ConcurrentHashMap<>();
```

Comment:
Use Map interface instead of implementation?

Reply:
Is this about the left-hand side of the `=`? I swear for the right-hand side, IDEA said that you can't do `new Map<>()` because it's an interface, much like in C#.
```suggestion
    private Map<World, RegionCoordinates> nextRegionCoordinates = new ConcurrentHashMap<>();
```

Reply:
Code suggestions? Wow I must've missed some GitHub blog update on PRs...

But ya, that.